### PR TITLE
Covarage ldap identity plugin

### PR DESCRIPTION
**Pull request by GustavoDinizMonteiro**

### CODE REVIEW

File: src/main/java/org/fogbowcloud/manager/core/plugins/behavior/federationidentity/ldap/LdapIdentityPlugin.java [hunk 1]
```diff
@@ -237,17 +236,16 @@ protected String ldapAuthenticate(String uid, String password) throws Exception
...
 
             ctx = new InitialDirContext(env);
 
             // Search the directory to get User Name and Domain from UID
             String filter = "(&(objectClass=inetOrgPerson)(uid={0}))";
             SearchControls ctls = new SearchControls();
-            ctls.setSearchScope(SearchControls.SUBTREE_SCOPE);
-            ctls.setReturningAttributes(new String[0]);
-            ctls.setReturningObjFlag(true);
+            ctls.setSearchScope(SearchControls.SUBTREE_SCOPE);
```
Comment (arn3tt) [inline src/main/java/org/fogbowcloud/manager/core/plugins/behavior/federationidentity/ldap/LdapIdentityPlugin.java]: Can you remove these additional whitespaces?

File: src/main/java/org/fogbowcloud/manager/core/plugins/behavior/federationidentity/ldap/LdapIdentityPlugin.java [hunk 2]
```diff
@@ -246,8 +244,8 @@ protected String ldapAuthenticate(String uid, String password) throws Exception
             String filter = "(&(objectClass=inetOrgPerson)(uid={0}))";
             SearchControls ctls = new SearchControls();
             ctls.setSearchScope(SearchControls.SUBTREE_SCOPE);
-            ctls.setReturningAttributes(new String[0]);
-            ctls.setReturningObjFlag(true);
+            ctls.setReturningAttributes(new String[0]);
```
Comment (arn3tt) [inline src/main/java/org/fogbowcloud/manager/core/plugins/behavior/federationidentity/ldap/LdapIdentityPlugin.java]: Can you remove these additional whitespaces?

File: src/test/java/org/fogbowcloud/manager/core/plugins/behavior/federationidentity/ldap/LdapIdentityPluginTest.java [hunk 3]
```diff
@@ -0,0 +1,171 @@
...
+		userCredentials.put(LdapIdentityPlugin.CRED_AUTH_URL, "ldapUrl");
+		userCredentials.put(LdapIdentityPlugin.CRED_LDAP_BASE, "ldapBase");
+		userCredentials.put(LdapIdentityPlugin.CRED_LDAP_ENCRYPT, "");
+		userCredentials.put(LdapIdentityPlugin.CRED_PRIVATE_KEY, "private_key_path");
+		userCredentials.put(LdapIdentityPlugin.CRED_PUBLIC_KEY, "public_key_path");
+	}
+	
+	@SuppressWarnings("unchecked")
+	@Test(expected = InvalidCredentialsUserException.class)
+	public void testCreateTokenWithoutCredentals() throws UnauthenticatedUserException, TokenValueCreationException {
```
Comment (thiagomanel) [inline src/test/java/org/fogbowcloud/manager/core/plugins/behavior/federationidentity/ldap/LdapIdentityPluginTest.java]: there is a typo here

File: src/test/java/org/fogbowcloud/manager/core/plugins/behavior/federationidentity/ldap/LdapIdentityPluginTest.java [hunk 4]
```diff
@@ -0,0 +1,171 @@
...
+		this.identityPlugin.createFederationTokenValue(Mockito.anyMap());
+	}
+	
+	
+	@Test
+	public void testLdapAuthenticateWithoutLdapUrl() throws Exception {
+		try {
+			this.identityPlugin.ldapAuthenticate(Mockito.anyString(), Mockito.anyString());
+			fail("Code above must trhow a exception");
```
Comment (thiagomanel) [inline src/test/java/org/fogbowcloud/manager/core/plugins/behavior/federationidentity/ldap/LdapIdentityPluginTest.java]: there is a typo here

File: src/test/java/org/fogbowcloud/manager/core/plugins/behavior/federationidentity/ldap/LdapIdentityPluginTest.java [hunk 5]
```diff
@@ -0,0 +1,171 @@
...
+		doReturn(userName).when(identityPlugin).ldapAuthenticate(Mockito.eq(name), Mockito.eq(password));
+		
+		String tokenA = identityPlugin.createFederationTokenValue(userCredentials);
+		
+		assertFalse(this.identityPlugin.isValid(tokenA));
+		
+	}
+	
+	@Test(expected = UnauthenticatedUserException.class)
+	public void testGetFederailsUserWithEmptyToken() throws UnauthenticatedUserException, UnexpectedException {
```
Comment (thiagomanel) [inline src/test/java/org/fogbowcloud/manager/core/plugins/behavior/federationidentity/ldap/LdapIdentityPluginTest.java]: typo here too

File: src/test/java/org/fogbowcloud/manager/core/plugins/behavior/federationidentity/ldap/LdapIdentityPluginTest.java [hunk 6]
```diff
@@ -0,0 +1,171 @@
...
+		this.identityPlugin.getFederationUser(Mockito.anyString());
+	}
+	
+	@Test(expected = UnauthenticatedUserException.class)
+	public void testGetFederailsUserWithInvalidToken() throws UnauthenticatedUserException, UnexpectedException {
```
Comment (thiagomanel) [inline src/test/java/org/fogbowcloud/manager/core/plugins/behavior/federationidentity/ldap/LdapIdentityPluginTest.java]: typo here

**Pull request merged**